### PR TITLE
[fix]: grpc state cleanup fix

### PR DESCRIPTION
**Pull request by oandreeva-nv**
#### What does the PR do?

This PR fixes intermittence of L0_grpc_state_cleanup.

2 sources of intermittence:

1) *_error_status: Was failing periodically because it was seeing `The stream is no longer in valid state, the error detail is reported through provided callback. A new stream should be started after stopping the current stream.`, but expecting `This protocol is restricted, expecting header 'triton-grpc-protocol-infer-key'`
Due to request count > 1, there is a time interval between when the request1 gets picked by gRPC server infer handler and it detect that the protocol was restricted. The requests sent in between this time period from client side will not see  "The stream is no longer in valid state" error.

It was decided to relax the condition and checking for both errors.

2) *_shutdownserver:

Was failing less frequent that 1 due to `Exception was not raised`. I believe something was happening to `os.kill` either being late in killing the process or some delay with killing process, so I've added a sleep after os.kill.

#### Checklist
- [X] PR title reflects the change and is of format `<commit_type>: <Title>`
- [X] Changes are described in the pull request.
- [ ] Related issues are referenced.
- [ ] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [ ] Added [test plan](#test-plan) and verified test passes.
- [X] Verified that the PR passes existing CI.
- [X] Verified copyright is correct on all changed files.
- [X] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [X] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [X] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [ ] test

#### Where should the reviewer start?
From the beginning

#### Test plan:
- CI Pipeline ID: ~16297581~ 16303863

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
- closes GitHub issue: #xxx

### CODE REVIEW

**Comment by tanmayv25** [comment]
> I believe something was happening to os.kill either being late in killing the process or some delay with killing process, so I've added a sleep after os.kill.

The test case is supposed to be testing that we cleanly handle state objects when the requests are received and cancelled when the server was under shutdown phase.
There might be some issues in server implementation during shutdown.

File: qa/L0_grpc_state_cleanup/cleanup_test.py [hunk 1]
```diff
@@ -428,11 +431,18 @@ def test_simple_infer_timeout(self):
     def test_simple_infer_error_status(self):
         # This test case is used to check whether all the state objects are
         # released when RPC runs into error.
+        expected_exceptions = [
+            "This protocol is restricted, expecting header 'triton-grpc-protocol-infer-key'",
+            "The stream is no longer in valid state, the error detail is reported through provided callback. A new stream should be started after stopping the current stream.",
```
Comment (GuanLuo) [inline qa/L0_grpc_state_cleanup/cleanup_test.py:436]: Is `_simple_infer` using gRPC streaming?

Comment (oandreeva-nv) [reply qa/L0_grpc_state_cleanup/cleanup_test.py:436]: good catch

**Comment by GuanLuo** [comment]
How can `This protocol is restricted, expecting header 'triton-grpc-protocol-infer-key'` break the stream state? Is that because the check is happening way early in the gRPC communication and the gRPC frontend does recover from it properly?

File: qa/L0_grpc_state_cleanup/cleanup_test.py (hunk 1 above)
Comment (oandreeva-nv) [reply qa/L0_grpc_state_cleanup/cleanup_test.py]: removed

**Comment by oandreeva-nv** [comment]
@tanmayv25 I've removed `time.sleep` and increased request count. In 12/12 tests exception was raised and intermittence seems to be no more. Could you please also help in answering @GuanLuo's question.

**Comment by tanmayv25** [comment]
>How can This protocol is restricted, expecting header 'triton-grpc-protocol-infer-key' break the stream state? Is that because the check is happening way early in the gRPC communication and the gRPC frontend does not recover from it properly?

The client is sending requests into a streaming connection with request delay 0. The requests that hit the protocol restricted error on the server gets returned as a [gRPC exception ](https://github.com/triton-inference-server/client/blob/main/src/python/library/tritonclient/grpc/_infer_stream.py#L157) which is caught in the client library. When error is caught, and the stream is marked [inactive](https://github.com/triton-inference-server/client/blob/44ed1215960e72cd241aef07ea6a33ce3d727265/src/python/library/tritonclient/grpc/_infer_stream.py#L162). 

The exception is caught and sent to the callback function for being handled by the client. The callback function just places the error into a queue which would be picked up by the client in future. However, the client still might send request after the stream is closed as the error object has not been picked from the queue. It is then we see this error appear. The gRPC frontend doesn't even register these requests.

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**